src/mux/tcp.rs

```diff
@@ -94,18 +101,22 @@ impl ChannelShared {
                     let mut protocol = protocol.borrow_mut();
                     match protocol.get_agency() {
                         Agency::Client => {
-                            let payload = protocol.send_data().unwrap();
-                            let id = protocol.protocol_id();
-                            let mut msg = Vec::new();
-                            msg.write_u32::<NetworkEndian>(self.start_time.elapsed().as_micros() as u32).unwrap();
-                            msg.write_u16::<NetworkEndian>(id).unwrap();
-                            msg.write_u16::<NetworkEndian>(payload.len() as u16).unwrap();
-                            msg.write(&payload[..]).unwrap();
-                            /* TODO:
-                             *   * Asynchronous Tx.
-                             *   * Handle errors.
-                             */
-                            self.stream.write(&msg).unwrap();
+                            match protocol.send_data() {
```

Comment:
Is this for outbound data verification?

Reply:
Sometimes we can have `Agency::Client`, but have no actual data to send and should receive an `Option::None` value back. In the case of TxSubmissionProtocol, if we have no transactions to submit, we just hang on to client agency until we do. We don't have to continually tell the server "I've got nothing for you", but instead just hang onto client agency until we have something to send them.